listView.php

```diff
@@ -351,6 +400,68 @@
+<div class="modal fade" id="addTask" tabindex="-1" aria-labelledby="addTask" aria-hidden="true">
+  <div class="modal-dialog">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h5 class="modal-title" id="exampleModalLabel">Add Task</h5>
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+          <span aria-hidden="true">&times;</span>
+        </button>
+      </div>
+      <div class="modal-body">
+        <form action="" class="addTask" method="POST">
+            <div class="form-group">
+                <label for="Task Title">Task Title</label>
+                <input style="height:50px;width:400px;font-size:20pt;" type="text" class="form-control" name="addTaskTitle" aria-describedby="emailHelp">
+                <small id="taskHelp" class="form-text text-muted">Give your task a descriptive title</small>
+            </div>
+            <div class="form-group">
+                <label for="taskDescription">Task Description</label>
+                <textarea style="height:150px;width:400px;font-size:14pt;"class="form-control" id="exampleFormControlTextarea1" rows="3" name="addTaskDetail"></textarea>
+                <small id="" class="form-text text-muted">Add a description with information regarding your task</small>
+            </div>
+            <!-- Retrieve existing list of task lists and display in a dropdown list for a user to select to add a task into-->
+            <div>
+            <label for="addListID"> Choose Task List</label>
+              <br>
+              <select name = "addListID">
+                  <option value="" disabled selected>Select Task List</option>
+                  <?php
+                      require("conn.php"); 
+                      $stmt = $db->prepare('SELECT listID,listTitle FROM taskList WHERE boardID=?');
+                          $stmt->execute(array($_GET['board']));       
+                          $taskLists = $stmt->fetchAll();
+                          foreach ($taskLists as $taskList){
+                              echo "<option value=" . $taskList['listID'] . ">" . $taskList['listTitle'] . "</option>"; 
+                          }
+                  ?> 
+              </select>
+            </div>
+            <div>
+              <label for="addTaskEffort"> Effort</label>
+              <input style="height:50px;width:80px;font-size:14pt;"type="number" class="form-control" name="addTaskEffort">
+              <small id="" class="form-text text-muted">Amount of work required</small>
+            </div>
+            <div>
+              <label for="addTaskTypeWork"> Type of Work</label>
+              <input style="height:50px;width:160px;font-size:14pt;"type="text" class="form-control" name="addTaskTypeWork">
+              <small id="" class="form-text text-muted">Detail of work type</small>
+            </div>
```

Comment:
Performed Unit Testing for the following cases(Effort Level and Type of Work):

- Null Inputs: The task doesn't get created, but there is no message prompting the user to add any input.
To fix this, I think there should be a `required` tag that can be added to make sure this doesn't happen, or add some default values in, just like for due date input

- Negative Input: It's accepting negative values for some reason, and also displaying it within the task attributes.
To fix this, I think there should be a lower bound(of 0) added to the inputs.

- Decimal Input: It's accepting decimal values as inputs too.
To fix this, I think the form can be made to only allow `int` datatypes as input, and display an error if not so.

Overall, I believe that for these inputs, rather than having free-range values, the user can be given different 'levels' as options(either in the form of radio buttons, check-boxes, or dropdown lists). This hasn't been discussed in the story though, so it isn't a requirement as of now. However, this can help to mitigate any chances of edge case errors as described above.

Reply:
• Null inputs: I will look into a default value. By design, a task will only be created when a task title and task list has been selected.
• Negative input: I agree with the suggested changes
• Decimal input: I agree with the suggested changes